docs/plugins.md

```diff
@@ -9,42 +9,66 @@ You will specifically want to review the docs:
 - [Topics](https://oclif.io/docs/topics)
 - [Plugins](https://oclif.io/docs/plugins)
 
-## 2. Clone the starter repo
+## 2. Adapt the starter repo (optional, but good for getting started)
+
+Start by cloning [this repo of an example debugger plugin](https://github.com/twilio/plugin-debugger).
+
+Once it's cloned, you can poke around to get a sense for the organization and even copy and adapt it for your own plugin use case.
+
+Your plugin directory should following the naming convention `plugin-<my-spectacular-plugin>`.
+
+Remember to update the `package.json` fields if you start building your own plugin off of this one:
+* name
+* description
+* homepage
+* oclif["name"]
+* topics
+
+And run `npm install`.
 
-Start by cloning [this repo of an example plugin](https://github.com/twilio/plugin-debugger).
 
 ## 3. Add your [topics](https://oclif.io/docs/topics) and [commands](https://oclif.io/docs/commands)
 
 Commands go in the `src/commands` folder and should inherit from one of our command base classes.
 
+### TwilioClientCommand
+
+Inherit from `TwilioClientCommand` if your command will need to make Twilio API calls. You will be provided with a `this.twilioClient` to make API calls using the Node.js helper library for Twilio. The client object will already have the necessary credentials and account SID. Just start calling the API. You are also given an `this.httpClient` to make requests to other API's, but you'll need to manage any necessary credentials yourself.
+
+### TwilioBaseCommand
+
+Inherit from `TwilioBaseCommand` if your command doesn't need to make any API calls.
+
+### Flags
+
+To create flags for your spectacular plugin, you will need the following import: `const { flags } = require('@oclif/command');`
+
+Additionally, you will need to copy the base class' flags:
+```javascript
+MySpectacularPlugin.flags = Object.assign({'fave-dessert': flag.string({'description': 'Your favorite dessert', required: true})}, TwilioClientCommand.flags)
+```
+
 ## 4. Test your plugin with the CLI
 
-Clone the main CLI repo (this repo) and [set it up](https://github.com/twilio/twilio-cli).
+Clone [the main CLI repo (this repo)](https://github.com/twilio/twilio-cli) and  set it up following the [twilio-cli Quickstart](https://www.twilio.com/docs/twilio-cli/quickstart).
```

Comment:
Should we have them install via npm or homebrew when developing/testing the plugin @dprothero ?

Reply:
Ah yes, I used npm yesterday!

Reply:
I think it should be npm since it's largely what you use for developing the plugin anyways